lib/jekyll-remote-theme/downloader.rb

```diff
@@ -37,31 +48,55 @@ def zip_file
         @zip_file ||= Tempfile.new([TEMP_PREFIX, ".zip"], :binmode => true)
       end
 
-      def download
-        Jekyll.logger.debug LOG_KEY, "Downloading #{zip_url} to #{zip_file.path}"
-        Net::HTTP.start(zip_url.host, zip_url.port, :use_ssl => true) do |http|
-          http.request(request) do |response|
-            raise_unless_sucess(response)
-            enforce_max_file_size(response.content_length)
-            response.read_body do |chunk|
-              zip_file.write chunk
+      def create_request(url)
+        req = Net::HTTP::Get.new url.path
+
+        remote_headers&.each do |key, value|
+          req[key] = value unless value.nil?
+        end
+
+        req
+      end
+
+      def handle_response(response)
+        raise_unless_success(response)
+        enforce_max_file_size(response.content_length)
+        response.read_body do |chunk|
+          zip_file.write chunk
+        end
+      end
+
+      def fetch(uri_str, limit = 10)
```

Comment:
Can you describe the change here? I'm having trouble following.

Reply:
It takes the remote_headers from the configuration file and ensures that the custom headers are passed to the request of the download of the remote_theme.  I had to reorganize the download function code for readability and usability.

More details are provided with additional comments of the relevant code.